src/domain/accounts/services/src/notifiers/account_lifecycle_notifier.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use askama::Template;
+use dill::{component, interface, meta, Catalog};
+use email_gateway::EmailSender;
+use internal_error::{InternalError, ResultIntoInternal};
+use kamu_accounts::{
+    AccountLifecycleMessage,
+    AccountLifecycleMessageCreated,
+    MESSAGE_CONSUMER_KAMU_ACCOUNTS_LIFECYCLE_NOTIFIER,
+    MESSAGE_PRODUCER_KAMU_ACCOUNTS_AUTHENTICATION_SERVICE,
+};
+use messaging_outbox::{
+    MessageConsumer,
+    MessageConsumerMeta,
+    MessageConsumerT,
+    MessageDeliveryMechanism,
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub const ACCOUNT_REGISTERED_SUBJECT: &str = "Welcome to Kamu!";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct AccountLifecycleNotifier {
+    email_sender: Arc<dyn EmailSender>,
+}
+
+#[component(pub)]
+#[interface(dyn MessageConsumer)]
+#[interface(dyn MessageConsumerT<AccountLifecycleMessage>)]
+#[meta(MessageConsumerMeta {
+    consumer_name: MESSAGE_CONSUMER_KAMU_ACCOUNTS_LIFECYCLE_NOTIFIER,
+    feeding_producers: &[
+        MESSAGE_PRODUCER_KAMU_ACCOUNTS_AUTHENTICATION_SERVICE,
+    ],
+    delivery: MessageDeliveryMechanism::Transactional,
+})]
+impl AccountLifecycleNotifier {
+    pub fn new(email_sender: Arc<dyn EmailSender>) -> Self {
+        Self { email_sender }
+    }
+
+    async fn notify_account_created(
+        &self,
+        created: &AccountLifecycleMessageCreated,
+    ) -> Result<(), InternalError> {
+        let registration_email = RegistrationEmail {
+            username: &created.display_name,
+        };
+        let rendered_registration_body = registration_email.render().unwrap();
+
+        self.email_sender
+            .send_email(
```

Comment:
Yea, as per my other comment this looks completely out of place in account domain crate.